core/src/test/java/org/elasticsearch/recovery/RelocationIT.java

```diff
@@ -351,7 +351,8 @@ public void testCancellationCleansTempFiles() throws Exception {
         client().admin().indices().prepareCreate(indexName)
                 .setSettings(Settings.builder().put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1, IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)).get();
 
-        internalCluster().startNodesAsync(2).get();
```

Comment:
it seems like we removed many usages of `startNodesAsync`. Does it still make sense to have that async method if it causes problems with how the test infra is setup by default?